scripts/process_image_dir.py

```diff
@@ -5,56 +5,80 @@
 from glob import glob
 from astropy.io import fits
 
+from pocs.utils import current_time
 from pocs.utils.config import load_config
 from pocs.utils.images import clean_observation_dir
 from pocs.utils.google.storage import upload_observation_to_bucket
 from pocs.utils.google.clouddb import add_header_to_db
 
 
-def main(directory, upload=True, send_headers=True, verbose=False):
+def main(directory, upload=True, send_headers=True, verbose=False, **kwargs):
     """Upload images from the given directory. """
 
+    def _print(msg):
+        if verbose:
+            print(msg)
+
     config = load_config()
     try:
         pan_id = config['pan_id']
     except KeyError:
         warn("Can't upload without a valid pan_id in the config")
         return
 
-    def _print(msg):
-        if verbose:
-            print(msg)
-
-    _print("Cleaning observation directory")
-    clean_observation_dir(directory, remove_jpgs=False)
+    _print("Cleaning observation directory: {}".format(directory))
+    try:
+        clean_observation_dir(directory, remove_jpgs=False, verbose=verbose, **kwargs)
+    except FileExistsError as e:
+        print(e)
 
     if upload:
         _print("Uploading to storage bucket")
-        upload_observation_to_bucket(pan_id, directory, include_files='*')
+
+        uploaded_files_fn = os.path.join(directory, 'uploaded_files.txt')
+
+        if os.path.exists(uploaded_files_fn):
+            _print("Files have been uploaded already, skipping")
+        else:
+            img_path = upload_observation_to_bucket(
```

Comment:
The name `img_path` implies this is the path of a single image. The `include_files='*'` implies that non-image files may be uploaded. Further, `img_path` is then passed to glob, implying that it is not a glob.

Reply:
I cleaned this up ab it but it's still not very intelligent. It merely records all files in the directory that were uploaded, assuming the upload was successful.